### PR TITLE
fix few bugs with workspace token

### PR DESCRIPTION
**Pull request by AidanAbd**

### CODE REVIEW

File: latch/resources/workflow.py [hunk 1]
```diff
@@ -98,6 +99,11 @@ def decorator(f: Callable):
                 raise click.exceptions.Exit(1)
 
         _inject_metadata(f, metadata)
-        return _workflow(f)
+
+        wf_name_override = os.environ.get("LATCH_WF_NAME_OVERRIDE")
```
Comment (ayushkamat) [inline latch/resources/workflow.py:103]: please add a comment saying this is only used for serializing inside the container - i guarantee ill be confused looking at this 6 months from now if not lol

File: latch_cli/constants.py [hunk 2]
```diff
@@ -35,6 +35,7 @@ class LatchConstants:
 
     pkg_name: str = "latch"
     pkg_config: str = ".latch/config"
+    pkg_workflow_name: str = ".latch/workflow_name"
```
Comment (ayushkamat) [inline latch_cli/constants.py:38]: can u put the wf_name in the config file instead

File: latch_cli/centromere/ctx.py [hunk 3]
```diff
@@ -396,7 +400,7 @@ def get_old_centromere_info(self) -> Tuple[str, str]:
             username = resp["username"]
         except KeyError as e:
             raise ValueError(
-                f"Malformed response from request for access token {resp}"
+                f"Malformed response from request to provision centromere {resp}"
```
Comment (ayushkamat) [inline latch_cli/centromere/ctx.py:403]: classic

File: latch_cli/services/register/register.py [hunk 4]
```diff
@@ -478,7 +486,7 @@ def register(
                         "The path to your provided dockerfile ",
                         f"{container.dockerfile} given to {task_name} is invalid.",
                     ) from e
-
+            print(protos)
```
Comment (ayushkamat) [inline latch_cli/services/register/register.py:489]: del

File: latch_cli/services/register/register.py [hunk 5]
```diff
@@ -428,6 +435,7 @@ def register(
                 local_td = Path(td)
 
             protos = _recursive_list(local_td)
+            print(protos)
```
Comment (ayushkamat) [inline latch_cli/services/register/register.py:438]: del

File: latch_cli/services/register/register.py [hunk 6]
```diff
@@ -225,6 +226,12 @@ def _build_and_serialize(
         image_build_logs, image_name, ctx.pkg_root, progress_plain=progress_plain
     )
 
+    # need to use
+    # from flytekit.models.admin import workflow as admin_workflow_models
+    # WorkflowSpec
+    # we can use from_flyte_idl to read it in and then patch with the proper target workflow
+    # probably have to do the same for launchplans
```
Comment (ayushkamat) [inline latch_cli/services/register/register.py:233]: del ?

**Review by ayushkamat — CHANGES_REQUESTED** [review]

File: latch/resources/workflow.py (hunk 1 above)
Comment (AidanAbd) [reply latch/resources/workflow.py:103]: amen lol

File: latch_cli/constants.py (hunk 2 above)
Comment (AidanAbd) [reply latch_cli/constants.py:38]: I tried, but it is really mess with how the config is being used (expected to be created in some places, expected to be not created in some places). Going to put in a separate file.

**Pull request merged**